util/file_reader.js

```diff
@@ -1,8 +1,44 @@
 import fs from "fs";
 import path from "path";
+import https from "https";
 
 const MAX_DEPTH = 2;
 
+export function determineFileSource(fileOrUrl, ext = null) {
+  // Currently assuming if we pass more than one param, they're *all* URLs
```

Comment:
hm, i think it should be flexible - if its a URL, we download to temporary file first, then we just load a list of files from disk..

Reply:
We would then need to combine this with loading files / folders that could be passed through the same param as well.